apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -13,6 +15,31 @@ import { showPublishDialog } from '../../templates/publishDialog/publishDialogRe
 import PublishDialog from '../../templates/publishDialog/PublishDialog';
 import { createHiddenPrintWindow } from '@cdo/apps/utils';
 import i18n from '@cdo/locale';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+
+function recordShare(type) {
+  if (!window.dashboard) {
+    return;
+  }
+
+  firehoseClient.putRecord({
+    study: 'finish-dialog-share',
+    study_group: 'v1',
+    event: 'project-share',
+    project_id: dashboard.project && dashboard.project.getCurrentId(),
+    data_string: type,
+  }, {includeUserId: true});
+}
+
+function wrapShareClick(handler, type) {
+  return function () {
+    try {
+      recordShare(type);
+    } finally {
```

Comment:
the try/finally does not look great -- normally if there were certain exceptions we were expecting, we would catch only those exceptions, and allow any unexpected exceptions to show up as JS errors in new relic. this pattern risks our reporting logic being broken without us noticing.

Reply:
`try` with no `catch` should still throw the error — it just ensures the `finally` block is run even if sending to Firehose throws for some reason.

![image](https://user-images.githubusercontent.com/413693/38146015-e77cd264-3400-11e8-836e-a99caccff1d6.png)

Reply:
Thanks for explaining, carry on!